rules/windows/process_creation/proc_creation_win_ads_stored_dll_execution_rundll32.yml

```diff
@@ -0,0 +1,22 @@
+title: Suspicious Alternate Data Stream Execution Via RunDLL32
+id: 9248c7e1-2bf3-4661-a22c-600a8040b446
+status: experimental
+description: Execution of DLL's stored in alternate data stream using RunDLL32
+references:
+    - https://lolbas-project.github.io/lolbas/Binaries/Rundll32
+author: Harjot Singh, '@cyb3rjy0t'
+date: 2023/01/11
+tags:
+    - attack.defense_evasion
+    - attack.t1564.004
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection:
+        Image|endswith: '\rundll32.exe'
+        CommandLine|re: 'rundll32.*\w:\\.+\..+:.+\.dll.*,\w+'
```

Comment:
A couple of notes to make sure of:

- I don't believe the DLL extension isn't required for the files. You can name it ".zzz" and it'll be fine (double check this)
- The path to the DLL can be directly the DLL itself (Example: "malware.dll") instead of a full path.
- There could be only spaces/double or single quotes between `rundll32` and the path so there is no need to use `.*`. Limit it to those three things maybe?
- There could be a space after the function call or it could be an ordinal...etc.

Reply:
Thanks for your comments.

I have reworked on regex and came with a query 

'[Rr][Uu][Nn][Dd][Ll][Ll]32.+\:.+\..+\:.+\..+\,.+$'

This would help detection of DLL execution without taking care of the extension name.

Reply:
I think you can even reduce it to something like `[Rr][Uu][Nn][Dd][Ll][Ll]32.+:` since the function call is optional.

Reply:
Well. I forget that you could also have a path in there. So let me think this through.

Reply:
To avoid writing complex regular expressions. We will settle for covering the generic case where a full path is used for the DLL at least in this first version.

Thanks for your contribution once again :D